synapse_rock/rockcraft.yaml

```diff
@@ -10,176 +10,219 @@ build-base: ubuntu@24.04
+    override-build: |
+      craftctl default
+      export RUSTUP_HOME=/rust
+      export CARGO_HOME=/cargo
+      export PATH=/cargo/bin:/rust/bin:$PATH
+      export CARGO_NET_GIT_FETCH_WITH_CLI=false
+      mkdir -p /rust /cargo /synapse /install
+      curl -m 30 -sSf https://sh.rustup.rs | sh -s -- -y --no-modify-path --default-toolchain=$RUST_VERSION --profile minimal
+      /rust/toolchains/$RUST_VERSION-x86_64-unknown-linux-gnu/bin/rustc -V > $CRAFT_PART_INSTALL/rust-version
+      pip3 install -U pip setuptools
+      pip3 install --root-user-action=ignore "poetry==$POETRY_VERSION"
+      cp pyproject.toml poetry.lock /synapse/
+      /usr/local/bin/poetry export --extras all -o /synapse/requirements.txt
+      pip3 install --prefix="/install" --no-deps --no-warn-script-location -r /synapse/requirements.txt
+      git clone https://github.com/canonical/synapse_stats_exporter.git
+      cd synapse_stats_exporter
+      pip3 install --prefix="/install" --no-warn-script-location -U .
+      cd ..
+      git clone --branch v1.4.0 --depth 1 https://github.com/matrix-org/synapse-s3-storage-provider
+      cd synapse-s3-storage-provider
+      pip3 install --prefix="/install" --no-warn-script-location -U .
+      cd ..
+      cp -r synapse /synapse/
+      cp -r rust /synapse/
+      cp pyproject.toml README.rst build_rust.py Cargo.toml Cargo.lock /synapse/
+      pip3 install --prefix="/install" --no-deps --no-warn-script-location /synapse[all];
+      git clone https://git.buechner.me/nbuechner/synapse-invite-checker
+      cd synapse-invite-checker
+      git checkout ab6c8b78d78c4cbf31e1a30981ae45c09285b34a
+      pip3 install --prefix="/install" --no-warn-script-location -U .
+      cd ..
+      # fix issue while creating file
+      # https://github.com/element-hq/synapse/issues/17882
+      pip3 install --prefix="/install" --force-reinstall -v "Twisted==24.7.0"
+      cp docker/start.py $CRAFT_PART_INSTALL/
+      chmod 755 $CRAFT_PART_INSTALL/start.py
+      cp -r docker/conf $CRAFT_PART_INSTALL/
+      cp -r /usr/local $CRAFT_PART_INSTALL/usr/
+      cp -r /install/local/* $CRAFT_PART_INSTALL/usr/local/
+      mkdir -p $CRAFT_PART_INSTALL/usr/local/attributemaps
+      chmod 755 $CRAFT_PART_INSTALL/usr/local/attributemaps
+      rm -rf $CRAFT_PART_INSTALL/aws/dist/awscli/examples/
+    overlay-packages:
+      - ca-certificates
+      - libjemalloc2
+  patches:
+    plugin: dump
+    after: [synapse]
+    source: patches
+    organize:
+      "*": srv/patches/
+  patch-start:
+    after:
+      - patches
+    plugin: nil
+    build-packages:
+      - git
+    override-stage: |
+      patch -f -u ./start.py -i srv/patches/start.patch
+  synapse-conf:
+    plugin: dump
+    source: attributemaps
+    organize:
+      login_ubuntu.py: /usr/local/attributemaps/login_ubuntu.py
+  mas-templates:
+    plugin: dump
+    <<: *mas-source
+    source-subdir: templates
+    organize:
+      "*": mas/share/templates/
```

Comment:
Out of curiosity: Is there a reason for dumping files in different parts?

Reply:
The main reason is I want to properly organize each component of MAS and give each of them their own part, this can also come in handy later if we want to do some patching of the translations or html templates